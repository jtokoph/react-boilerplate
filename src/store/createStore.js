import { applyMiddleware, compose, createStore as createReduxStore } from 'redux';
import thunk from 'redux-thunk';
import createRootReducer from '../reducers/root';

const createStore = (initialState = {}) => {
  // ======================================================
  // Middleware Configuration
  // ======================================================
  const middleware = [thunk];

  // ======================================================
  // Store Enhancers
  // ======================================================
  const enhancers = [];
  let composeEnhancers = compose;

  /* istanbul ignore if */
  if (__DEV__) {
    if (typeof window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ === 'function') { // eslint-disable-line no-underscore-dangle
      composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__; // eslint-disable-line no-underscore-dangle,max-len
    }
  }

  // ======================================================
  // Store Instantiation and HMR Setup
  // ======================================================
  const store = createReduxStore(
    createRootReducer(),
    initialState,
    composeEnhancers(
      applyMiddleware(...middleware),
      ...enhancers,
    ),
  );

  store.asyncReducers = {};

  /* istanbul ignore if */
  if (module.hot) {
    module.hot.accept('../reducers/root', () => {
      const reducers = require('../reducers/root').default; // eslint-disable-line
      store.replaceReducer(reducers(store.asyncReducers));
    });
  }

  return store;
};

export default createStore;
