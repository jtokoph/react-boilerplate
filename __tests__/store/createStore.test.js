import createStore from '../../src/store/createStore';

describe('store', () => {
  const store = createStore();

  test('should be an object', () => {
    expect(typeof store).toBe('object');
  });

  test('should be a proper store', () => {
    expect(store).toHaveProperty('dispatch');
    expect(store).toHaveProperty('getState');
  });

  test('should have proper default state', () => {
    expect(store.getState()).toMatchSnapshot();
  });
});
