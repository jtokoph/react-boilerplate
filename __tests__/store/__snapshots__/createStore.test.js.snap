// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store should have proper default state 1`] = `
Object {
  "counter": 0,
}
`;
